docs/tutorials/friends_lobbies.md

```diff
@@ -0,0 +1,104 @@
+# Tutorials - Getting a list of lobbies friends are in
+
+In this quick tutorial we will cover how to get a dictionary of lobbies to friends that are visible to the user. This was created as it's a pretty common feature for multiplayer games.
+
+We will break down the different steam functions to call first before looking at the code samples.
+
+---
+
+## Functions explained
+
+```gdscript
+var number_of_friends: int = Steam.getFriendCount(Steam.FRIEND_FLAG_IMMEDIATE)
+```
+This must be ran before `Steam.getFriendByIndex` according to the Steamworks documentation, we'll be using this for our loop over each friend in our friends list.
+
+```gdscript
+var steam_id: int = Steam.getFriendByIndex(index, Steam.FRIEND_FLAG_IMMEDIATE)
+```
+Inside of the loop we'll call `Steam.getFriendByIndex` to get the Steam ID of our friend at this index.
+
+Note: you must use the same flag as the flag used inside of `Steam.getFriendCount`.
+
+```gdscript
+var game_info: Dictionary = Steam.getFriendGamePlayed(steam_id)
+```
+With our friends Steam ID we can now get information about the game our friend may or may not be playing.
+
+This dictionary will be empty if the friend is offline or not playing a game.
+
+Here's the fields and possible values inside this dictionary:
+
+- **id**: int
+  - The app id of the game they are playing.
+- **lobby**: int ***or*** String
+  - If the friend is in a lobby visible to this user, it will be an int of the lobby ID
+  - Otherwise, it will be a String equal to: `"No valid lobby"`
+- **ip**: String
+- **game_port**: int
+- **query_port**: int
+  - Depends on the lobby settings, but `ip`, `game_port`, and `query_port` will probably all just be zeros.
+
+---
+
+## Putting it all together
+
+```gdscript
+# get a dictionary of lobbies with friends
+# key-value pair: lobby_id -> Array<steam_id>
+func get_lobbies_with_friends() -> Dictionary:
+  var results = {}
+  var num_of_friends: int = Steam.getFriendCount()
+  for i in range(0, num_of_friends):
+    var steam_id: int = Steam.getFriendByIndex(i, Steam.FRIEND_FLAG_IMMEDIATE)
+    var game_info: Dictionary = Steam.getFriendGamePlayed(steam_id)
+    if game_info.has("lobby"):
+      var app_id: int = game_info.id
+      var lobby = game_info.lobby
+      if app_id != Steam.getAppID() or lobby is String:
+        continue
+      if not results.has(lobby):
+        results[lobby] = []
+      results[lobby].append(steam_id)
+  return results
```

Comment:
I feel this code could be made simpler/easier to parse, and to be more in line with the other tutorials; have you considered:

```
func get_lobbies_with_friends() -> Dictionary:
  var results: Dictionary = {}

  for i in range(0, Steam.getFriendCount()):
    var steam_id: int = Steam.getFriendByIndex(i, Steam.FRIEND_FLAG_IMMEDIATE)
    var game_info: Dictionary = Steam.getFriendGamePlayed(steam_id)
  
    if not 'lobby' in game_info:
        continue # This friend was not online, or not in a game
    else: # They were in a lobby, check if it's the same game as ours
        var app_id: int = game_info['id']
        var lobby = game_info['lobby']
        if app_id != Steam.getAppID() or lobby is String:
          continue # Either not in this game, or not in a lobby
      
        if not results.has(lobby):
          results[lobby] = []

        results[lobby].append(steam_id)
  return results
  ```

Reply:
Thanks for the suggestion! The game info being empty isn't really about the steam id being valid or not, it can be empty if the friend isn't playing a game or is offline. I'll update the comment as such.